src/openms/source/METADATA/ID/IdentificationData.cpp

```diff
@@ -628,126 +717,128 @@ namespace OpenMS
                        });
     }
 
-    // remove molecule-query matches based on identified molecules:
-    set<IdentifiedMoleculeRef> id_refs;
-    for (auto it = identified_peptides_.begin();
+    // remove observation matches based on identified molecules:
+    set<IdentifiedMolecule> id_vars;
+    for (IdentifiedPeptideRef it = identified_peptides_.begin();
          it != identified_peptides_.end(); ++it)
     {
-      id_refs.insert(it);
+      id_vars.insert(it);
     }
-    for (auto it = identified_compounds_.begin();
+    for (IdentifiedCompoundRef it = identified_compounds_.begin();
          it != identified_compounds_.end(); ++it)
     {
-      id_refs.insert(it);
+      id_vars.insert(it);
     }
-    for (auto it = identified_oligos_.begin();
+    for (IdentifiedOligoRef it = identified_oligos_.begin();
          it != identified_oligos_.end(); ++it)
     {
-      id_refs.insert(it);
+      id_vars.insert(it);
     }
-    removeFromSetIf_(query_matches_, [&](MoleculeQueryMatches::iterator it)
+    removeFromSetIf_(observation_matches_, [&](ObservationMatches::iterator it)
                      {
-                       return !id_refs.count(it->identified_molecule_ref);
+                       return !id_vars.count(it->identified_molecule_var);
```

Comment:
potential performance problem? count will go through the whole thing

Reply:
"The whole thing" is a set, so `count` will just test for existence.